src/plugin/tests/hooks.spec.ts

```
@@ -0,0 +1,510 @@
+  on: vi.fn(),
+  off: vi.fn(),
+  _eventListeners: new Map<string, Function[]>(),
+  _preventEventLoops: true, // Flag to prevent test event loops
+};
+
+// Add controlled event emission to prevent infinite loops
+mockPluginManager.on = vi.fn((event: string, callback: Function) => {
+  if (!mockPluginManager._eventListeners.has(event)) {
+    mockPluginManager._eventListeners.set(event, []);
+  }
+  mockPluginManager._eventListeners.get(event)!.push(callback);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)